pkg/virtctl/expose/expose.go

```diff
@@ -0,0 +1,213 @@
+        A virtual machine replica set will be exposed as a service only if its selector is convertible to a selector that service supports, i.e. when the selector contains only the matchLabels component.
+        Note that if no port is specified via --port and the exposed resource has multiple ports, all will be re-used by the new service. 
+        Also if no labels are specified, the new service will re-use the labels from the resource it exposes.
+        
+        Possible types are (case insensitive, both single and plurant forms):
+        
+        virtualmachine (vm), offlinevirtualmachine (ovm), virtualmachinereplicaset (vmrs)`,
+		Example: usage(),
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Command{command: COMMAND_EXPOSE, clientConfig: clientConfig}
+			return c.RunE(cmd, args)
+		},
+	}
+
+	// flags for the "expose" command
+	cmd.Flags().StringVar(&serviceName, "name", "", "Name of the service created for the exposure of the VM")
+	cmd.MarkFlagRequired("name")
+	cmd.Flags().StringVar(&clusterIP, "cluster-ip", "", "ClusterIP to be assigned to the service. Leave empty to auto-allocate, or set to 'None' to create a headless service.")
+	cmd.Flags().StringVar(&externalIP, "external-ip", "", "Additional external IP address (not managed by the cluster) to accept for the service. If this IP is routed to a node, the service can be accessed by this IP in addition to its generated service IP. Optional.")
+	cmd.Flags().StringVar(&loadBalancerIP, "load-balancer-ip", "", "IP to assign to the Load Balancer. If empty, an ephemeral IP will be created and used.")
+	cmd.Flags().Int32Var(&port, "port", 0, "The port that the service should serve on")
+	cmd.MarkFlagRequired("port")
+	cmd.Flags().StringVar(&strProtocol, "protocol", "TCP", "The network protocol for the service to be created.")
+	cmd.Flags().IntVar(&intTargetPort, "target-port", 0, "Name or number for the port on the VM that the service should direct traffic to. Optional.")
+	cmd.Flags().Int32Var(&nodePort, "node-port", 0, "Port used to expose the service on each node in a cluster.")
+	cmd.Flags().StringVar(&strServiceType, "type", "ClusterIP", "Type for this service: ClusterIP, NodePort, or LoadBalancer.")
+	cmd.Flags().StringVar(&portName, "port-name", "", "Name of the port. Optional.")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+
+	return cmd
+}
+
+func usage() string {
+	usage := "  # Expose SSH to a virtual machine called 'myvm' as a node port (5555) of the cluster:\n"
+	usage += fmt.Sprintf("  virtctl expose vm myvm --port=5555 --target-port=22 --name=myvm-ssh --type=NodePort")
+	return usage
+}
+
+// executing the "expose" command
+func (o *Command) RunE(cmd *cobra.Command, args []string) error {
+	// first argument is type of VM: VM, offline VM or replica set VM
+	vmType := strings.ToLower(args[0])
```

Comment:
Can you use cobra's positional arguments instead?

Reply:
maybe I'm missing something, but according to this: https://github.com/spf13/cobra#positional-and-custom-arguments I think I'm using positional arguments (calling: ```cobra.ExactArgs(2)```)?

Reply:
Yes. Arguably we should in all our commands use custom validators, to not just return the generic error message.

Reply:
so, we should probably keep it like this, unless we decide to re-do all commands?